script-test/playbackstrategies/modifiers/live/restartabletest.js

```diff
@@ -120,13 +109,93 @@ require(
             it('playFrom', function () {
               isUndefined('playFrom');
             });
+          });
+
+          describe('should use fake time for', function () {
+            var timeUpdates = [];
+            function timeUpdate (opts) {
+              timeUpdates.forEach(function (fn) { fn(opts); });
+            }
+
+            beforeEach(function () {
+              jasmine.clock().install();
+              jasmine.clock().mockDate();
+
+              player.addEventCallback.and.callFake(function (self, callback) {
+                timeUpdates.push(callback);
+              });
+
+              initialiseRestartableMediaPlayer();
+            });
+
+            afterEach(function () {
+              jasmine.clock().uninstall();
+            });
+
+            describe('getCurrentTime', function () {
+              it('should be set on to the window length on beginPlayback', function () {
+                restartableMediaPlayer.beginPlayback();
+
+                expect(restartableMediaPlayer.getCurrentTime()).toBe(100);
+              });
+
+              it('should start at supplied time', function () {
+                restartableMediaPlayer.beginPlaybackFrom(10);
+
+                expect(restartableMediaPlayer.getCurrentTime()).toBe(10);
+              });
+
+              it('should increase when playing', function () {
+                restartableMediaPlayer.beginPlaybackFrom(10);
+
+                timeUpdate({ state: MediaPlayerBase.STATE.PLAYING });
 
-            it('getCurrentTime', function () {
-              isUndefined('getCurrentTime');
+                jasmine.clock().tick(1000);
+
+                timeUpdate({ state: MediaPlayerBase.STATE.PLAYING });
+
+                expect(restartableMediaPlayer.getCurrentTime()).toBe(11);
+              });
+
+              it('should not increase when paused', function () {
+                restartableMediaPlayer.beginPlaybackFrom(10);
+                timeUpdate({ state: MediaPlayerBase.STATE.PAUSED });
+
+                jasmine.clock().tick(1000);
+
+                timeUpdate({ state: MediaPlayerBase.STATE.PLAYING });
+
+                expect(restartableMediaPlayer.getCurrentTime()).toBe(10);
+              });
             });
 
-            it('getSeekableRange', function () {
-              isUndefined('getSeekableRange');
+            describe('getSeekableRange', function () {
+              it('should start at the window time', function () {
+                restartableMediaPlayer.beginPlaybackFrom(0);
+
+                timeUpdate({ state: MediaPlayerBase.STATE.PLAYING });
+
+                expect(restartableMediaPlayer.getSeekableRange()).toEqual({ start: 0, end: 100 });
+              });
+
+              it('should increase with time', function () {
```

Comment:
Could the test name explicitly call out that this is for sliding window (as the test below calls it out for growing)